File: 0066-plus-one/0066-plus-one.cpp
class Solution {
public:
    vector<int> plusOne(vector<int>& arr) {
        arr[arr.size()-1]++;
        int carry = 0;
        int sum = 0;
        for(int i=arr.size()-1;i>=0;i--){
            // if(i==arr.size()-1)sum = 1 + arr[i];
            sum = carry + arr[i];
            if(sum > 9){
                carry = 1;
                arr[i] = sum - 10;
            }
            else{
                carry = 0;
                arr[i] = sum;
            }
        }
        if(carry)arr.insert(arr.begin(),carry);
        return arr;

    }
};

File: 0066-plus-one/0066-plus-one.java
class Solution {
    public int[] plusOne(int[] arr) {
        int carry = 1, sum =0;
        ArrayList <Integer> ans = new ArrayList<>();
        for(int i = arr.length -1; i>=0;i--){
            sum =carry + arr[i];
            int digit  = sum%10;
            ans.add(0,digit);
            carry = sum>9?1:0;
        }
        if(carry==1)ans.add(0,1);
        int [] ans1 =new int[ans.size()];
        for(int i=0;i<ans.size();i++)ans1[i] = ans.get(i);
        return ans1;
    }
}